tools/build_defs/framework.bzl

```diff
@@ -404,23 +429,29 @@ def _define_inputs(attrs):
     linking_infos_all = []
 
     bazel_headers = []
+    bazel_system_includes = []
     bazel_libs = []
 
     # This framework function-built libraries: copy result directories under
     # $EXT_BUILD_DEPS/lib-name
     ext_build_dirs = []
+    ext_build_dirs_set = {}
 
     for dep in attrs.deps:
-        provider = dep[OutputGroupInfo]
-        ext_built = provider and hasattr(provider, "gen_dir")
+        provider = dep[ExternallyBuiltTransitive]
 
         linking_infos_all += [dep[CcLinkingInfo]]
         compilation_infos_all += [dep[CcCompilationInfo]]
 
-        if ext_built:
-            ext_build_dirs += provider.gen_dir.to_list()
+        if provider:
+            for artifact in provider.artifacts:
+                if not ext_build_dirs_set.get(artifact.gen_dir):
+                    ext_build_dirs_set[artifact.gen_dir] = 1
+                    ext_build_dirs += [artifact.gen_dir]
         else:
```

Comment:
In what case would there not be a provider?

Reply:
It is a bit unreadable, sorry.
I was referring to the explanation above: we are iterating the transitive depset of dependencies, which in general can contain duplicates (some libraries B and C in "upper fragment" of a dependency chain can use the same "leaf" library A).

There would not be a provider in case the dependency is not build with this foreign_cc_rules rule, and is just usual Bazel cc_library (or something compatible). In that case we should act according to know Bazel interface - use library, use include directory.